### DISCUSSION TITLE
git clone' command the error 'port 22:  Network is unreachable'

### DISCUSSION THREAD

**Discussion by AYurov28** [community/community#41435] [New to GitHub]
Hello everyone!
I am trying to clone the Spoon-Knife repository, and when I use the 'git clone' command the error 'port 22:  Network is unreachable' keeps coming up:

********
$ git clone git@github.com:AYurov28/Spoon-Knife.git
Cloning into 'Spoon-Knife'...
ssh: connect to host github.com port 22: Network is unreachable
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
*******

I have tried a number of different ways to solve this problem including turning off my firewall but nothing seems to be working.  I would be grateful if anybody who has encountered the same issue could help me. 
Thank you!

**Comment by queenofcorgis** [comment]
Have you been able to clone any other repositories?

**Reply by th-sakib** [reply]
I have same issue and I can't clone any repository using ssh.  But I am able to clone with https link...

**Comment by AYurov28** [comment]
Hello Lili,

The answer is no. I am new to GitHab and it is my first bite at cloning the repos.

I manage to establish connection via HTTPS route. However cant get SSH route working.

I understand that the issue might be with the firewalls configuration given that the error mentions port22. Any advice how to resolve it woud be gratefully appreciated.

Thank you
Artem
________________________________

**Comment by joe-crawford** [comment]
We are also seeing intermittent failures to connect to GitHub. Connecting to other websites works OK.

Example from today:
```sh
$ ssh -Tvvv github.com 22
OpenSSH_7.6p1 Ubuntu-4ubuntu0.7, OpenSSL 1.0.2n  7 Dec 2017
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: Applying options for *
debug2: resolving "github.com" port 22
debug2: ssh_connect_direct: needpriv 0
debug1: Connecting to github.com [140.82.121.4] port 22.
debug1: connect to address 140.82.121.4 port 22: Connection timed out
ssh: connect to host github.com port 22: Connection timed out
```

Retrying later worked.

**Comment by airtower-luna** [comment]
@AYurov28, if connections to `github.com` on other ports work it sounds a lot like a firewall *somewhere* along the path (could be on your computer, but doesn't have to be) is blocking your packets. The other likely problem based on the error message is a routing problem, but if you can reach GitHub on other ports that's unlikely, unless you're using a proxy for HTTPS.

**Comment by AYurov28** [comment]
Thank you.


  1.  I understand the below firewal status report (from comand prompt) suggest that Port22 is open, but I still receive the error "port22:Network is unreachable. Could not read from remote repository."

  1.  When I try to use HTTPS route  I manage to clone using HTTPS but when I try to apply command "git comit -m" in Git Bach  (to commit changes applied in VSC) there is no response and Git Bash just shows ">" sign on the next row with no following actions. Are HTTPS and SSH routes differ at this stage? May I ask if there are documents/videos explaining SSH and HTTPS  workflows in details?

[cid:c563f6eb-ad21-4107-b693-369bfaf8b3e9]

Thank you.
Artem
________________________________

**Comment by airtower-luna** [comment]
> I understand the below firewal status report (from comand prompt) suggest that Port22 is open, but I still receive the error "port22:Network is unreachable. Could not read from remote repository."

I don't see any firewall status in your message. As I wrote above, it doesn't have to be the firewall *on your computer*. It could be some router along the way that's blocking SSH. I don't see why anyone would do that for outgoing connections, but I've seen so many misguided "security" policies that I wouldn't be surprised either. :sweat_smile: 

On Linux you could use the following command to figure out where your connection gets rejected, for other systems you'd have to figure out the equivalent:

```
traceroute -T -p 22 github.com
```

The options are to make traceroute use TCP (`-T`) and try to connect to port 22.

> When I try to use HTTPS route  I manage to clone using HTTPS but when I try to apply command "git comit -m" in Git Bach  (to commit changes applied in VSC) there is no response and Git Bash just shows ">" sign on the next row with no following actions.

That actually looks like something is wrong with the command you typed (maybe a missing closing quote), but without seeing the exact command I can't say for sure. What I can say for sure is that `git commit` doesn't care about the protocol you use for a remote, because it won't make any network connection. That will only happen when you try to push the new commit. :wink:

**Comment by mwpowellhtx** [comment]
Getting similar errors. I can push changes to other repositories, so at least on some level, I know there is connectivity.

Did verify the `traceroute` thingy, seem to have a fairly quick responsive route, so it seems at least as though the _port 22_ issue can be ruled out as the culprit.

```
$ git clone git@github.com:.../....git .../
Cloning into '...'...
socket: No buffer space available
ssh: connect to host github.com port 22: failure
fatal: Could not read from remote repository.

Please make sure you have the correct access rights and the repository exists.
```

Names and repos blipped for brevity.

Should I start by updating my Windows git? Been a minute since I last did that, but I cannot imagine why the underlying request would be an issue now.

Possibly an out of date SSH? Need to update that maybe?

Open to suggestions what's going on there.

**Reply by mwpowellhtx** [reply]
Okay, verified in an actual Ubuntu prompt, not the `git` prompt... starting to wonder if I should just go ahead and update my `git`.

```
'''@''':/mnt/x/Users/...$ eval "$(ssh-agent -s)"
Agent pid ###
```

Identifying details clipped for brevity.

**Reply by mwpowellhtx** [reply]
Okay, so thus far seems as though maybe the issue is refreshing my `.ssh` keys, making sure those are all aligned, from W10 WSL2 linux instances, to git ssh bookkeeping, and so on.
